agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/hdfs/FileSystemTracer.java

```diff
@@ -311,23 +297,34 @@ public static Object intercept(
             Method getServerAddress = getMethod(classLoader, "org.apache.hadoop.ipc.RPC", "getServerAddress", Object.class);
             InetSocketAddress inetSocketAddress = (InetSocketAddress) getServerAddress.invoke(o, rpcProxy);
             return executeMethod(zuper, "hdfs://" + inetSocketAddress.getHostString() + ":" + inetSocketAddress.getPort(),
-                    null, dst, FsAction.ADD_BLOCK.name());
+                    null, dst, FsAction.ADD_BLOCK.name(), null);
         }
     }
 
-    private static Object executeMethod(@SuperCall Callable<?> zuper, String uri, String src, String dst, String fsAction) throws Exception {
+    private static Object callDistributedFileSystem(@SuperCall Callable<?> zuper, @This Object o, String src, String dst, String fsAction) throws Exception {
+        ClassLoader classLoader = o.getClass().getClassLoader();
+        Field dfsField = getField(classLoader, "org.apache.hadoop.hdfs.DistributedFileSystem", "dfs");
```

Comment:
Any reason for not using something like .class.getCanonicalName() instead of strings? (I'm guessing maybe because we don't want to depend on these packages)

Reply:
Please look at #65